### PR TITLE
added missing if none match header when reading nodes or edges

### PR DESCRIPTION
**Pull request by hkernbach**

### CODE REVIEW

File: js/client/tests/http/api-gharial-spec.js [hunk 1]
```diff
@@ -608,4 +607,116 @@ describe('_api/gharial', () => {
...
+    expect(req.statusCode).to.equal(200);
+    expect(req.json.edge).to.deep.equal(doc);
+  });
+
+  it('should check if the if-match header is working - negative', () => {
+    const examples = require('@arangodb/graph-examples/example-graph');
+    const exampleGraphName = 'knows_graph';
+    const vName = 'persons';
+    const eName = 'knows';
+    expect(db._collection(eName)).to.be.null; // edgec
```
Comment (jsteemann) [inline js/client/tests/http/api-gharial-spec.js]: does it have to be `to.be.null()` (i.e. with parentheses)??

Comment (hkernbach) [reply js/client/tests/http/api-gharial-spec.js]: expect(null).to.be.null; // Recommended

found in http://www.chaijs.com/api/bdd/

Comment (jsteemann) [reply js/client/tests/http/api-gharial-spec.js]: Ok!

**Comment by hkernbach** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/670/

**Pull request merged**